api/tests/mocha/services/report/smsparser.spec.js

```diff
@@ -97,10 +97,82 @@ describe('sms parser', () => {
     chai.expect(smsparser.getFormCode('द CDT33')).to.equal('द');
   });
 
+  it('accepts form with phone_number type', () => {
+    const doc = { message: 'NP 20 9841202020' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal('+9779841202020');
+  });
+
+  it('accepts phone number with extension', () => {
+    const doc = { message: 'NP 20 +9779841202020' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal('+9779841202020');
+  });
+
+  it('accepts correct phone number without extension', () => {
+    const doc = { message: 'NP 20 9841202020' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal('+9779841202020');
```

Comment:
Isn't 3rd test the exact duplicate of first test here ?

Reply:
Hmm. Yes. Any ideas on how I can distinguish testing phone_number field is supported i an different way or the second test should suffice?

@yrimal